app/src/main/java/com/github/swent/echo/compose/components/ListDrawer.kt

```diff
@@ -23,126 +28,254 @@ import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Size
 import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import com.github.swent.echo.R
+import com.github.swent.echo.data.SAMPLE_EVENTS
 import com.github.swent.echo.data.model.Event
+import com.github.swent.echo.viewmodels.STATUS_THRESHOLD
 import java.time.format.DateTimeFormatter
 
 @Composable
 fun ListDrawer(eventsList: List<Event>) {
     val selectedEvent = remember { mutableStateOf("") }
+    // Main column where every items will be displayed, scrollable
     LazyColumn(
         modifier =
             Modifier.fillMaxSize().background(Color.White).padding(5.dp).testTag("list_drawer")
     ) {
+        // Iterate over the list of events and display them
         items(eventsList) { event ->
-            EventListItem(event = event, selectedEvent = selectedEvent)
+            EventListItem(event = event, selectedEvent = selectedEvent, onJoinButtonPressed = {})
             Spacer(modifier = Modifier.height(5.dp))
         }
     }
 }
 
 @Composable
-fun EventListItem(event: Event, selectedEvent: MutableState<String>) {
+fun EventListItem(
+    event: Event,
+    selectedEvent: MutableState<String>,
+    onJoinButtonPressed: () -> Unit
+) {
+    // Colors for the background of the list item
+    val fraction1 = 0.8f
+    val fraction2 = 0.6f
+    val listColor1 = MaterialTheme.colorScheme.primaryContainer
+    val listColor2 =
+        listColor1.copy(
+            red = listColor1.red * fraction1,
+            green = listColor1.green * fraction1,
+            blue = listColor1.blue * fraction1
+        )
+    val listColor3 =
```

Comment:
I assume that the variations of colors is for aesthetic purposes but do not forget to comment and explain as to what their functionality is in their code.

Reply:
yes giving them better name would be more understandable, I will do that in the futur cleanup